File: app/src/main/java/pl/vemu/zsme/detailedNews/galleryFragment/GalleryPageAdapter.java
package pl.vemu.zsme.detailedNews.galleryFragment;

import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.github.chrisbanes.photoview.PhotoView;

import coil.Coil;
import coil.request.ImageRequest;

import static android.view.ViewGroup.LayoutParams.MATCH_PARENT;

public class GalleryPageAdapter extends RecyclerView.Adapter<GalleryPageAdapter.ViewHolder> {

    private final String[] images;
    private static SwitchUIVisibility switchUIVisibility;

    public static void setSwitchUIVisibility(SwitchUIVisibility switchUIVisibility) {
        GalleryPageAdapter.switchUIVisibility = switchUIVisibility;
    }

    public GalleryPageAdapter(String[] images) {
        this.images = images;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        PhotoView photoView = new PhotoView(parent.getContext());
        photoView.setLayoutParams(new ViewGroup.LayoutParams(MATCH_PARENT, MATCH_PARENT));
        return new ViewHolder(photoView);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        ImageRequest request = new ImageRequest.Builder(holder.photoView.getContext())
                .data(images[position])
                .crossfade(true)
                .target(holder.photoView)
                .build();
        Coil.imageLoader(holder.photoView.getContext()).enqueue(request);
    }

    @Override
    public int getItemCount() {
        return images.length;
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {

        private final PhotoView photoView;

        public ViewHolder(@NonNull PhotoView itemView) {
            super(itemView);
            this.photoView = itemView;
            this.photoView.setOnClickListener(view -> switchUIVisibility.switchUiVisibility());
        }
    }
}

File: app/src/main/java/pl/vemu/zsme/ui/GalleryPageAdapter.kt
package pl.vemu.zsme.ui

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.ViewGroup.LayoutParams.MATCH_PARENT
import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import coil.load
import com.github.chrisbanes.photoview.PhotoView

class GalleryPageAdapter(private val fragment: GalleryFragment, private val images: Array<String>) : FragmentStateAdapter(fragment) {

    override fun getItemCount() = images.size

    override fun createFragment(position: Int): Fragment = GalleryPageFragment(fragment, images[position])

}

class GalleryPageFragment(private val fragment: GalleryFragment, private val image: String) : Fragment() {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View =
            PhotoView(context).apply {
                layoutParams = ViewGroup.LayoutParams(MATCH_PARENT, MATCH_PARENT)
                load(image)
                setOnClickListener { fragment.switchUiVisibility() }
            }
}